### ISSUE TITLE
CA1062 False Positive when null coalescing operator used with String.Empty

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2582]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3

#### Diagnostic ID

[CA1062: Validate arguments of public methods](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps

1. Enable CA1062
2. Analyze the following code:

```csharp
public class Class1
{
    public Class1(int num) { }

    public Class1(string name)
        : this((name ?? string.Empty).Length) // CA1062 here!
    {
    }
}
```

#### Expected behavior
No warning, since `string.Empty` is never `null`. This warning was NOT reported using the legacy/binary FxCop.

#### Actual behavior
CA1062 warning

**Comment by mavasani** [comment]
https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19317.5+45537f7f should have the fix

**Comment by RobSiklos** [comment]
Verified fixed in https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19318.2+bc3ca30e

Thanks!
